test/Parse/collection_conditionals_error.swift

```diff
@@ -0,0 +1,60 @@
+// RUN: %target-typecheck-verify-swift
+
+[ // expected-warning {{expression of type '[String]' is unused}}
+  #if FLAG
+  "foo",
+  #else
+  "foo": 12, // expected-error {{expected ',' separator}}
+  // expected-error@-1{{expected expression in container literal}}
+  #endif
+]
+
+[ // expected-warning {{expression of type '[String : Int]' is unused}}
+#if FLAG
+  /* "foo": 1000, */
+#else
+  "bar": 42,
+#endif
+  "baz": 0,
+]
+
+[
+#if true
+] // expected-error {{expected expression in container literal}}
+#endif // expected-error {{unexpected conditional compilation block terminator}}
+
+[ // expected-warning {{expression of type '[Int]' is unused}}
+#if true
+  42 // expected-error {{expected ',' separator}}
+#endif
+]
+
+[ // expected-warning {{expression of type '[Int]' is unused}}
+#if true
+  42 // expected-error {{expected ',' separator}}
+#endif
+  , // expected-error {{expected expression in container literal}}
+  12,
+]
+
+_ = [
+  42,
+#if true
+  12, 2 // expected-error {{expected ',' separator}}
+#endif
+  0
+]
+
+_ = [
+#if true
+  "foo": 42 // expected-error {{expected ',' separator}}
+#endif
+]
+
+_ = [
+  1 // expected-error {{expected ',' separator}}
+#if // expected-error {{incomplete condition in conditional compilation directive}}
```

Comment:
Sorry I meant `#if true` or something. There's no need to test "missing condition" here.

Reply:
Then it works fine. I’ve pushed an amend.